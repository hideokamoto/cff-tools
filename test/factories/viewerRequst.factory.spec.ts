import { TestRequestEventFactory } from '../../src/factories/viewerRequest.factory';

describe('factories/viewerRequest.factory.ts', () => {
  describe('TestRequestEventFactory', () => {
    it('should return default event json', () => {
      const builder = TestRequestEventFactory.create();
      expect(builder.getEventObject()).toEqual(
        JSON.stringify({
          version: '1.0',
          context: {
            eventType: 'viewer-request',
          },
          viewer: {
            ip: '',
          },
          request: {
            method: 'GET',
            uri: '/index.html',
            querystring: {},
            headers: {},
            cookies: {},
          },
        })
      );
    });
    it('should match snapshot', () => {
      const builder = TestRequestEventFactory.create();
      builder
        .setRequestContext({
          distributionId: 'EXXX',
        })
        .setViewerIP('127.0.0.0')
        .setRequestMethod('POST')
        .setRequestUri('admin/index.html')
        .setRequestQuerystring({
          test1: {
            value: '123',
          },
          test2: {
            value: '222',
            multiValue: [
              {
                value: '222',
              },
              {
                value: '333',
              },
            ],
          },
        })
        .setRequestHeaders({
          host: {
            value: 'www.example.com',
          },
          accept: {
            value: 'text/html',
            multiValue: [
              {
                value: 'text/html',
              },
              {
                value: 'application/xhtml+xml',
              },
            ],
          },
        })
        .setRequestCookies({
          id: {
            value: 'CookieIdValue',
          },
          loggedIn: {
            value: 'false',
          },
        });
      expect(builder.getEvent()).toMatchSnapshot();
    });
  });
});
