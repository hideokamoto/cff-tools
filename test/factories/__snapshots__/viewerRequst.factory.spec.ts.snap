// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`factories/viewerRequest.factory.ts TestRequestEventFactory should match snapshot 1`] = `
Object {
  "context": Object {
    "eventType": "viewer-request",
  },
  "request": Object {
    "cookies": Object {
      "id": Object {
        "value": "CookieIdValue",
      },
      "loggedIn": Object {
        "value": "false",
      },
    },
    "headers": Object {
      "accept": Object {
        "multiValue": Array [
          Object {
            "value": "text/html",
          },
          Object {
            "value": "application/xhtml+xml",
          },
        ],
        "value": "text/html",
      },
      "host": Object {
        "value": "www.example.com",
      },
    },
    "method": "POST",
    "querystring": Object {
      "test1": Object {
        "value": "123",
      },
      "test2": Object {
        "multiValue": Array [
          Object {
            "value": "222",
          },
          Object {
            "value": "333",
          },
        ],
        "value": "222",
      },
    },
    "uri": "admin/index.html",
  },
  "version": "1.0",
  "viewer": Object {
    "ip": "127.0.0.0",
  },
}
`;
